geode-core/src/main/java/org/apache/geode/internal/cache/wan/GatewaySenderEventImpl.java

```diff
@@ -1179,6 +1190,38 @@ public Long getShadowKey() {
     return this.shadowKey;
   }
 
+  public boolean equals(Object obj) {
+    if (this == obj) {
+      return true;
+    }
+
+    if (obj == null || !(obj instanceof GatewaySenderEventImpl)) {
```

Comment:
I think the null check is redundant here

Reply:
Thanks for the review. I'll remove that check and also add a unit test for this method.